src/main/java/seedu/address/logic/commands/ChargeCommand.java

```diff
@@ -81,7 +83,39 @@ public CommandResult execute(Model model) throws CommandException {
 
         model.updateFilteredPetList();
         return new CommandResult(generateSuccessMessage(petToCharge.getName(),
-                amountChargeable, getMonthName()));
+                amountChargeable, getMonthName(), chargeDate.getYear(), getAttendance(petToCharge)));
+    }
+
+    /**
+     * Returns the attendance the charge on the pet is based on
+     * @return attendance as a String.
+     */
+    public String getAttendance(Pet pet) {
+        double amountChargeable = 0.0;
+        String attendance = "\nHere are the days " + pet.getName().toString() + " was present: ";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        AttendanceHashMap map = pet.getAttendanceHashMap();
+        LocalDate day = this.chargeDate.atDay(1);
+        LocalDate firstDayNextMonth = this.chargeDate.atEndOfMonth().plusDays(1);
+        while (day.isBefore(firstDayNextMonth)) {
+            boolean hasAttendance = map.getAttendance(day).isPresent();
+            if (hasAttendance) {
+                AttendanceEntry entry = map.getAttendance(day).get();
+                boolean isPresent = entry.getIsPresent().orElse(false);
+                if (isPresent) {
+                    amountChargeable += this.charge.getCharge();
+                    attendance += "\n" + entry.getAttendanceDate().format(formatter);
+                }
+            }
+            day = day.plusDays(1);
+        }
```

Comment:
I think this part is a bit too nested! Could benefit from more refactoring, depending on how you would like to organise your code.

Reply:
good suggestion, I think I will extract the inner-if statement to a boolean check to reduce nesting!